platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -338,4 +338,29 @@ public static AddressBook buildAddressBook(@NonNull final Roster roster) {
 
         return addressBook;
     }
+
+    /**
+     * Verifies that all nodes are the same between the two specified rosters.
+     *
+     * @param roster1 the first roster to compare
+     * @param roster2 the second roster to compare
+     * @throws IllegalStateException if the rosters do not match
+     */
+    public static void verifyReconnectRosters(@NonNull final Roster roster1, @NonNull final Roster roster2)
+            throws IllegalStateException {
+        final int numNodes = roster1.rosterEntries().size();
+
+        if (numNodes != roster2.rosterEntries().size()) {
+            throw new IllegalStateException("Rosters have different number of nodes");
+        }
+
+        for (int i = 0; i < numNodes; ++i) {
+            final RosterEntry roster1Entry = roster1.rosterEntries().get(i);
+            final RosterEntry roster2Entry = roster2.rosterEntries().get(i);
+
+            if (!roster1Entry.equals(roster2Entry)) {
```

Comment:
If this `.equals()` check is sufficient for comparing `RosterEntry`, then the `.equals()` check is sufficient on rosters.   I think the whole method could be collapsed to:
```
return roster1.equals(roster2);
```
@anthony-swirldslabs Thoughts?

Reply:
If this is sufficient, then the whole `verifyReconnectRoster()` method can go away and be replaced by a roster equality check.